src/XrdHttp/XrdHttpProtocol.cc

```diff
@@ -446,6 +448,23 @@ static long BIO_XrdLink_ctrl(BIO *bio, int cmd, long num, void * ptr)
   case BIO_CTRL_FLUSH:
     ret = 1;
     break;
+  case BIO_C_SET_NBIO:
```

Comment:
I see this addition but I don't see any use of it. Could you point me to the code that uses this case?

Reply:
Look around line 730 in this file -- `BIO_set_nbio(sbio, 1);`.

`SSL_peek_ex` relies on the BIO implementing non-blocking IO to function (well, to function without blocking).